generic/utils/ras.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+import os
+import commands
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class RASTools(Test):
+
+    def setUp(self):
+        Arch = commands.getoutput("arch")
+        if "ppc" not in Arch:
+            self.skip("supported only on Power platform")
+        sm = SoftwareManager()
+        if not sm.check_installed("ppc64-diag") and not sm.install("ppc64-diag"):
+            self.error('ppc64-diag is needed for the test to be run')
+        if not sm.check_installed("powerpc-utils") and not sm.install("powerpc-utils"):
+            self.error('powerpc-utils is needed for the test to be run')
+        if not sm.check_installed("lsvpd") and not sm.install("lsvpd"):
+            self.error('lsvpd is needed for the test to be run')
+
+    def test(self):
+        scriptToRun = self.params.get("script", default=None)
+        toolToRun = self.params.get("tool", default=None)
+        dir = os.path.join(self.datadir, 'log')
+        if not os.path.exists(dir):
+            os.makedirs(dir)
+        commandToRun = './ras.py.data/' + scriptToRun + ' ' + \
+            '>>' + ' ' + 'ras.py.data/log/output.txt' + ' ' + '2>&1'
```

Comment:
Get the full path for the script and output file, not the relative one, it fails


2016-06-10 19:46:13,945 multiplexer      L0182 DEBUG| PARAMS (key=script, path=*, default=None) => 'sys_ident_tool.sh'
2016-06-10 19:46:13,945 multiplexer      L0182 DEBUG| PARAMS (key=tool, path=*, default=None) => 'sys_ident'
2016-06-10 19:46:13,949 output           L0632 DEBUG| Return code is 256
2016-06-10 19:46:13,950 process          L0331 INFO | Running './ras.py.data/sys_ident_tool.sh >> ras.py.data/log/output.txt 2>&1'
2016-06-10 19:46:13,955 stacktrace       L0036 ERROR|
2016-06-10 19:46:13,955 stacktrace       L0039 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado-35.0-py2.7.egg/avocado/core/test.py:435
2016-06-10 19:46:13,960 stacktrace       L0042 ERROR| Traceback (most recent call last):
2016-06-10 19:46:13,960 stacktrace       L0042 ERROR|   File "/var/lib/libvirt/images/sath/avocado-wrapper-final-1/tests/avocado-misc-tests/generic/utils/ras.py", line 35, in test
2016-06-10 19:46:13,960 stacktrace       L0042 ERROR|     cmd_result = process.run(commandToRun, ignore_status=True)
2016-06-10 19:46:13,960 stacktrace       L0042 ERROR|   File "/usr/lib/python2.7/site-packages/avocado-35.0-py2.7.egg/avocado/utils/process.py", line 1028, in run
2016-06-10 19:46:13,960 stacktrace       L0042 ERROR|     cmd_result = sp.run(timeout=timeout)
2016-06-10 19:46:13,960 stacktrace       L0042 ERROR|   File "/usr/lib/python2.7/site-packages/avocado-35.0-py2.7.egg/avocado/utils/process.py", line 556, in run
2016-06-10 19:46:13,961 stacktrace       L0042 ERROR|     self._init_subprocess()
2016-06-10 19:46:13,961 stacktrace       L0042 ERROR|   File "/usr/lib/python2.7/site-packages/avocado-35.0-py2.7.egg/avocado/utils/process.py", line 346, in _init_subprocess
2016-06-10 19:46:13,961 stacktrace       L0042 ERROR|     raise exc
2016-06-10 19:46:13,961 stacktrace       L0042 ERROR| OSError: File './ras.py.data/sys_ident_tool.sh' not found
2016-06-10 19:46:13,961 stacktrace       L0043 ERROR|
2016-06-10 19:46:13,965 test             L0550 ERROR| Traceback (most recent call last):

2016-06-10 19:46:13,965 test             L0550 ERROR|   File "/usr/lib/python2.7/site-packages/avocado-35.0-py2.7.egg/avocado/core/test.py", line 490, in _run_avocado
    raise test_exception